OPHD/UI/DetailMap.cpp

```diff
@@ -66,6 +66,7 @@ void DetailMap::resize(NAS2D::Vector<int> size)
 	// Set up map draw position
 	const auto sizeInTiles = size.skewInverseBy(TileSize);
 	mMapView.viewSize(std::min(sizeInTiles.x, sizeInTiles.y));
+	mMapView.validateViewLocation();
```

Comment:
I think the check should be incorportated into the `viewSize` method. When you change the view size, you may need to adjust the position to keep the view from going off the bottom or right edge of the map.

The original code:
```cpp
void MapView::viewSize(int sizeInTiles)
{
	mEdgeLength = std::max(3, sizeInTiles);
}
```

I think after the `mEdgeLength` value is updated, a call should be added:
```
mapViewLocation(mOriginTilePosition);
```
That will cause the map view position to be re-clipped, taking into account the new `mEdgeLength`.